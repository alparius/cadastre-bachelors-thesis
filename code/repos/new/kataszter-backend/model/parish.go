package model

import "go.mongodb.org/mongo-driver/bson/primitive"

type Parish struct {
	ID               primitive.ObjectID `json:"ID,omitempty" bson:"_id,omitempty"`
	Name             string             `json:"name,omitempty" bson:"name,omitempty"`
	CityName         string             `json:"cityName,omitempty" bson:"cityName,omitempty"`
	Verified         bool               `json:"verified,omitempty" bson:"verified,omitempty"`
	Description      string             `json:"description,omitempty" bson:"description,omitempty"`
	History          string             `json:"history,omitempty" bson:"history,omitempty"`
	PastNames        []PastNames        `json:"pastNames,omitempty" bson:"pastNames,omitempty"`
	Coordinates      Coordinates        `json:"coordinates,omitempty" bson:"coordinates,omitempty"`
	AssistantPriests []ParishPlacement  `json:"assistantPriests,omitempty" bson:"assistantPriests,omitempty"`
	MainPriests      []ParishPlacement  `json:"mainPriests,omitempty" bson:"mainPriests,omitempty"`
	Pictures         []string           `json:"pictures,omitempty" bson:"pictures,omitempty"`
	Files            []string           `json:"files,omitempty" bson:"files,omitempty"`
}
