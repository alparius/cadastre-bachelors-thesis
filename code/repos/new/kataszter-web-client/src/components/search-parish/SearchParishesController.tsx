import React from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { Form, Grid, Header } from "semantic-ui-react";

import IParishFilters from "../../data/parish/ParishFilters";
import BoldFM from "../../util/styledMessages/BoldFM";

interface ISearchParishesControllerProps {
    filters: IParishFilters;
    isListView: boolean;
    setFilters(newFilters: object): void;
    toggleMap(): void;
    resetPagination(): void;
}

const SearchParishesController: React.FC<ISearchParishesControllerProps> = (props: ISearchParishesControllerProps) => {
    const { filters, isListView, setFilters, toggleMap, resetPagination } = props;
    const { formatMessage: fm } = useIntl();

    const handleChange = (event: any) => {
        const { name, value } = event.target;
        setFilters({ ...filters, [name]: value });
        resetPagination();
    };

    return (
        <Form>
            <Header as="h1" className="controllerHeader">
                <FM id="search.parishes" />
            </Header>

            <Grid>
                <Grid.Row unstackable style={{ marginBottom: "1.5em", marginTop: "1.5em" }} columns={2}>
                    <Grid.Column textAlign="right">
                        <BoldFM id="label.listView" />
                    </Grid.Column>
                    <Grid.Column textAlign="left">
                        <Form.Checkbox id="listView" name="toggleListView" toggle checked={isListView} onChange={toggleMap} />
                    </Grid.Column>
                </Grid.Row>
            </Grid>
            <Form.Input id="name" name="parishName" label={fm({ id: "label.name" })} value={filters.parishName} onChange={handleChange} />
            <Form.Input id="city" name="cityName" label={fm({ id: "label.parish.cityName" })} value={filters.cityName} onChange={handleChange} />
            <Form.Input
                name="minSize"
                type="number"
                label={fm({ id: "search.minSize" })}
                value={filters.minSize ? Number(filters.minSize) : undefined}
                onChange={handleChange}
            />
        </Form>
    );
};

export default SearchParishesController;
