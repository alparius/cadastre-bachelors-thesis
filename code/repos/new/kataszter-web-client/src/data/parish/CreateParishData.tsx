interface ICreateParishData {
    name: string;
    cityName: string;
}

export let NewCreateParishData: ICreateParishData = {
    name: "",
    cityName: ""
};

export default ICreateParishData;
