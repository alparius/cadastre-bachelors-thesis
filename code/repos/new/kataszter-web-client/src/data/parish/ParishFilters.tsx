interface IParishFilters {
    parishName: string;
    cityName: string;
    minSize: string;
}

export let NewParishFilters: IParishFilters = {
    parishName: "",
    cityName: "",
    minSize: ""
};

export default IParishFilters;
