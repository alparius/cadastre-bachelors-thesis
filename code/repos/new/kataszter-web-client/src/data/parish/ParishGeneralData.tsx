import ICoordinates, { NewCoordinates } from "../Coordinates";
import IParishPastName, { NewParishPastName } from "./ParishPastName";
import IParishPlacement, { NewParishPlacement } from "./ParishPlacement";

interface IParishGeneralData {
    ID: number;
    name: string;
    cityName: string;
    verified: boolean;
    description: string;
    history: string;
    pastNames: IParishPastName[];
    coordinates: ICoordinates;
    assistantPriests: IParishPlacement[];
    mainPriests: IParishPlacement[];
    pictures: string[];
    files: string[];
}

export let NewParishGeneralData: IParishGeneralData = {
    ID: 0,
    name: "",
    cityName: "",
    verified: false,
    description: "",
    history: "",
    pastNames: [NewParishPastName],
    coordinates: NewCoordinates,
    assistantPriests: [NewParishPlacement],
    mainPriests: [NewParishPlacement],
    pictures: [],
    files: []
};

export default IParishGeneralData;
