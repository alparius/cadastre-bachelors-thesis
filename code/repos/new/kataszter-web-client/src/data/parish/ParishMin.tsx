import ICoordinates, { NewCoordinates } from "../Coordinates";

interface IParishMin {
    ID: number;
    name: string;
    verified: boolean;
    numberOfPriests: string;
    cityName: string;
    coordinates: ICoordinates;
}

export let NewParishMin: IParishMin = {
    ID: 0,
    name: "",
    verified: true,
    numberOfPriests: "",
    cityName: "",
    coordinates: NewCoordinates
};

export default IParishMin;
