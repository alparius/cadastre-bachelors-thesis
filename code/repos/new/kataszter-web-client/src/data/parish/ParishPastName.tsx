import IDate, { NewDate } from "../Date";

interface IParishPastName {
    name: string;
    startDate: IDate;
    endDate: IDate;
}

export let NewParishPastName: IParishPastName = {
    name: "",
    startDate: NewDate,
    endDate: NewDate
};

export default IParishPastName;
