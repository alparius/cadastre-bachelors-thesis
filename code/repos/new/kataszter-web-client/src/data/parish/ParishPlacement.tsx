import IDate, { NewDate } from "../Date";

interface IParishPlacement {
    ID?: number;
    priestID: number;
    priestName: string;
    startDate: IDate;
    endDate: IDate;
    numberOfPeople: string;
}

export let NewParishPlacement: IParishPlacement = {
    priestID: 0,
    priestName: "",
    startDate: NewDate,
    endDate: NewDate,
    numberOfPeople: ""
};

export default IParishPlacement;
