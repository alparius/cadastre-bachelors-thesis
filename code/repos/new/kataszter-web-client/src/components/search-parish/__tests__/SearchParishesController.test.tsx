/*===================
  SearchParishesController.unit.test.tsx
 ===================*/
import React from "react";

import { NewParishFilters } from "../../../data/parish/ParishFilters";
import { createComponentWithIntl, mountWithIntl } from "../../../util/test/test-wrapper";
import SearchParishesController from "../SearchParishesController";

const handlers = {
    setFilters: jest.fn(),
    resetPagination: jest.fn(),
    toggleMap: jest.fn()
};

describe("SearchParishesController element", () => {
    afterEach(() => {
        handlers.setFilters.mockClear();
        handlers.resetPagination.mockClear();
    });

    it("should render the form with search input fields", () => {
        const container = createComponentWithIntl(<SearchParishesController filters={NewParishFilters} {...handlers} isListView />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the handlePriestNameChange function when the #name input field changes", () => {
        const container = mountWithIntl(<SearchParishesController filters={NewParishFilters} {...handlers} isListView />);
        const input = container.find("input#name");
        input.simulate("change", { target: { value: "te" } });
        input.simulate("change", { target: { value: "st" } });
        expect(handlers.setFilters).toHaveBeenCalledTimes(2);
    });
});
