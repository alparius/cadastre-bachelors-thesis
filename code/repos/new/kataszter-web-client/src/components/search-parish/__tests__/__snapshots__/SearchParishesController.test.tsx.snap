// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchParishesController element should render the form with search input fields 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <h1
    className="ui header controllerHeader"
  >
    Search parishes
  </h1>
  <div
    className="ui grid"
  >
    <div
      className="two column row"
      style={
        Object {
          "marginBottom": "1.5em",
          "marginTop": "1.5em",
        }
      }
      unstackable={true}
    >
      <div
        className="right aligned column"
      >
        <b>
          List view
        </b>
      </div>
      <div
        className="left aligned column"
      >
        <div
          className="field"
        >
          <div
            className="ui checked fitted toggle checkbox"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              checked={true}
              className="hidden"
              id="listView"
              name="toggleListView"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label
              htmlFor="listView"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="ui input"
    >
      <input
        id="name"
        name="parishName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="city"
    >
      Town
    </label>
    <div
      className="ui input"
    >
      <input
        id="city"
        name="cityName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Minimum number of priests
    </label>
    <div
      className="ui input"
    >
      <input
        name="minSize"
        onChange={[Function]}
        type="number"
      />
    </div>
  </div>
</form>
`;
